src/components/split_button/split_button.tsx

```diff
@@ -0,0 +1,289 @@
+      openPopover();
+    }
+  });
+
+  const setItemNode = (node: HTMLButtonElement | null, index: number) => {
+    itemNodes[index] = node;
+  };
+
+  const closePopover = () => {
+    setIsOpen(false);
+  };
+
+  const itemClicked = (index: number) => {
+    setIsOpen(false);
+    if (onChange) {
+      onChange(index);
+    }
+  };
+
+  const onSelectKeyDown = (event: React.KeyboardEvent<HTMLButtonElement>) => {
+    if (event.key === keys.ARROW_UP || event.key === keys.ARROW_DOWN) {
+      event.preventDefault();
+      event.stopPropagation();
+      openPopover();
+    }
+  };
+
+  const onItemKeyDown = (event: React.KeyboardEvent<HTMLButtonElement>) => {
+    switch (event.key) {
+      case keys.ESCAPE:
+        // close the popover and prevent ancestors from handling
+        event.preventDefault();
+        event.stopPropagation();
+        closePopover();
+        break;
+
+      case keys.TAB:
+        // no-op
+        event.preventDefault();
+        event.stopPropagation();
+        break;
+
+      case keys.ARROW_UP:
+        event.preventDefault();
+        event.stopPropagation();
+        shiftFocus(ShiftDirection.BACK);
+        break;
+
+      case keys.ARROW_DOWN:
+        event.preventDefault();
+        event.stopPropagation();
+        shiftFocus(ShiftDirection.FORWARD);
+        break;
+    }
+  };
+
+  const focusItemAt = (index: number) => {
+    const targetElement = itemNodes[index];
+    if (targetElement != null) {
+      targetElement.focus();
+    }
+  };
+
+  const shiftFocus = (direction: ShiftDirection) => {
+    const currentIndex = itemNodes.indexOf(
+      document.activeElement as HTMLButtonElement
+    );
+    let targetElementIndex: number;
+
+    if (currentIndex === -1) {
+      // somehow the select options has lost focus
+      targetElementIndex = 0;
+    } else {
+      if (direction === ShiftDirection.BACK) {
+        targetElementIndex =
+          currentIndex === 0 ? itemNodes.length - 1 : currentIndex - 1;
+      } else {
+        targetElementIndex =
+          currentIndex === itemNodes.length - 1 ? 0 : currentIndex + 1;
+      }
+    }
+
+    focusItemAt(targetElementIndex);
+  };
+
+  const popoverClasses = classNames('ouiSplitButton', popoverClassName);
+
+  const buttonClasses = classNames(
+    {
+      'ouiSplitButton--isOpen__button': isOpen,
+    },
+    className
+  );
+
+  const itemClasses = classNames(
+    'ouiSplitButton__item',
+    {
+      'ouiSplitButton__item--hasDividers': hasDividers,
+    },
+    itemClassName
+  );
+
+  const button = (
+    <OuiSplitButtonControl
+      onDropdownClick={isOpen ? closePopover : openPopover}
+      onClick={onClick}
+      onKeyDown={onSelectKeyDown}
+      className={buttonClasses}
+      fullWidth={fullWidth}
+      // compressed={compressed}
+      {...rest}>
+      {children}
+    </OuiSplitButtonControl>
+  );
+
+  const items = options.map((option, index) => {
+    const isSelected = selectedIndexInt === index;
+
```

Comment:
@BSFishy Could you tell me how I could resolve the typedef for this color?

```
error TS2322: Type '"text" | "success" | "accent" | "warning" | "danger" | "ghost"' is not assignable to type '"default" | "success" | "secondary" | "accent" | "warning" | "danger" | "subdued" | "ghost" | undefined'.
  Type '"text"' is not assignable to type '"default" | "success" | "secondary" | "accent" | "warning" | "danger" | "subdued" | "ghost" | undefined'.
```

I have created OuiSplitButtonColor as a compromise... still not good enough for tsc.